test/Microsoft.AspNetCore.Http.Connections.Tests/HttpConnectionDispatcherTests.cs

```diff
@@ -1616,6 +1616,93 @@ public TrackingMemoryStream(SemaphoreSlim callerTracker, Task waitTask)
+                connection.TransportType = transportType;
+                connection.Items[ConnectionMetadataNames.Transport] = transportType;
+
+                var dispatcher = new HttpConnectionDispatcher(manager, loggerFactory);
+
+                var context = MakeRequest("/foo", connection);
+                SetTransport(context, transportType);
+
+                var serviceCollection = new ServiceCollection();
+                serviceCollection.AddSingleton<TestConnectionHandler>();
+                var services = serviceCollection.BuildServiceProvider();
+                var builder = new ConnectionBuilder(services);
+                builder.UseConnectionHandler<TestConnectionHandler>();
+                var app = builder.Build();
+                var options = new HttpConnectionOptions();
+
+                _ = dispatcher.ExecuteAsync(context, options, app).OrTimeout();
+
+                // Issue the delete request
+                var deleteContext = new DefaultHttpContext();
+                deleteContext.Request.Path = "/foo";
+                deleteContext.Request.QueryString = new QueryString($"?id={connection.ConnectionId}");
+                deleteContext.Request.Method = "DELETE";
+                var ms = new MemoryStream();
+                deleteContext.Response.Body = ms;
+
+                await dispatcher.ExecuteAsync(deleteContext, options, app).OrTimeout();
+
+                // Verify the response from the DELETE request
+                Assert.Equal(StatusCodes.Status400BadRequest, deleteContext.Response.StatusCode);
+                Assert.Equal("text/plain", deleteContext.Response.ContentType);
+                Assert.Equal("Cannot terminate this connection using the DELETE endpoint.", Encoding.UTF8.GetString(ms.ToArray()));
+            }
+        }
+
+        [Fact]
+        public async Task DeleteEndpointGracefullyTerminatesLongPolling()
+        {
+            using (StartLog(out var loggerFactory, LogLevel.Debug))
+            {
+                var manager = CreateConnectionManager(loggerFactory);
+                var connection = manager.CreateConnection();
+                connection.TransportType = HttpTransportType.LongPolling;
+                connection.Items[ConnectionMetadataNames.Transport] = HttpTransportType.LongPolling;
+
+                var dispatcher = new HttpConnectionDispatcher(manager, loggerFactory);
+
+                var context = MakeRequest("/foo", connection);
+
+                var services = new ServiceCollection();
+                services.AddSingleton<TestConnectionHandler>();
+                var builder = new ConnectionBuilder(services.BuildServiceProvider());
+                builder.UseConnectionHandler<TestConnectionHandler>();
+                var app = builder.Build();
+                var options = new HttpConnectionOptions();
+
+                var pollTask = dispatcher.ExecuteAsync(context, options, app).OrTimeout();
```

Comment:
Do you want the timeout here instead of on the awaiting of the pollTask itself?

Reply:
We usually do it this way, but I do think it fits better on the `await`. I can put it there.

Reply:
Yah